### PR TITLE
Sync param check for DiskANN and HNSW

### PR DESCRIPTION
**Pull request by liliu-z**
This PR is to sync how HNSW&DiskANN will perform with empty config with Knowhere 1.x:

The rule is:
1. If `ef` exists, we need to ensure `ef` >= `k`.
2. If `ef` not exists, let `ef` = max(16, `k`).

### CODE REVIEW

File: src/index/hnsw/hnsw_config.h [hunk 1]
```diff
@@ -15,11 +15,15 @@
...
 
-// This is not a valid EF value for HNSW
-// This value is used to tell if HnswConfig.ef is coming from user or not
-const int64_t kDefaultHnswEfPlaceholder = -1;
-
 namespace knowhere {
+
+namespace {
+
+constexpr const CFG_INT::value_type kDefaultEfDivider = 16;
```
Comment (cydrain) [inline src/index/hnsw/hnsw_config.h:22]: what "Divider" mean ?

File: src/index/hnsw/hnsw_config.h [hunk 2]
```diff
@@ -15,11 +15,15 @@
...
+constexpr const CFG_INT::value_type kDefaultRangeSearchEf = 16;
```
Comment (cydrain) [inline src/index/hnsw/hnsw_config.h:23]: can we use one "kDefaultEf" for both search and range search ?

Comment (liliu-z) [reply src/index/hnsw/hnsw_config.h:23]: They don't have to be the same

File: src/index/hnsw/hnsw_config.h (hunk 1 above)
Comment (liliu-z) [reply src/index/hnsw/hnsw_config.h:22]: meaning if k < divider, ef = divider. Anyway, more than welcome to better name, I agree it is poor

**Pull request merged**